x/mongo/driver/topology/topology_options.go

```diff
@@ -333,6 +335,28 @@ func NewConfig(co *options.ClientOptions, clock *session.ClusterClock) (*Config,
 		)
 	}
 
+	if opts := co.LoggerOptions; opts != nil {
+		if opts == nil {
+			opts = options.Logger()
+		}
+
+		// Build an internal component-level mapping.
+		componentLevels := make(map[logger.Component]logger.Level)
+		for component, level := range opts.ComponentLevels {
+			componentLevels[logger.Component(component)] = logger.Level(level)
+		}
+
+		log, err := logger.New(opts.Sink, opts.MaxDocumentLength, componentLevels)
+		if err != nil {
+			return nil, fmt.Errorf("error creating logger: %v", err)
```

Comment:
Wrap the returned error instead.
```suggestion
			return nil, fmt.Errorf("error creating logger: %w", err)
```

Reply:
Historically, we seem to use `%v` for errors throughout our code. Is there a way that we can add a linter rule for it?

Reply:
We could add [polyfloyd/go-errorlint](https://github.com/polyfloyd/go-errorlint#fmterrorf-wrapping-verb) to our golangci-lint, but, as you note, we historically always use `%v` so we would probably have to update quite a bit of code to use that linter effectively.